gammapy/scripts/cta_irf.py

```diff
@@ -0,0 +1,105 @@
+from ..background import Cube
+from ..irf import EnergyDispersion2D
+from ..irf import EnergyDependentMultiGaussPSF
+
+__all__ = [
+    'CTAIrf',
+]
+
+
+class CTAIrf(object):
+    """Contain CTA instrument response functions.
+
+    Class handling CTA instrument response function. For now
+    we use the production 2 of the CTA IRF
+    (https://portal.cta-observatory.org/Pages/CTA-Performance.aspx)
+    adapted from the ctools
+    (http://cta.irap.omp.eu/ctools/user_manual/getting_started/response.html).
+    The IRF format should be complient compliant with the one discussed
+    at http://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/.
+    Waiting for a new public production of the CTA IRF,
+    we'll fix the missing pieces.
+
+    Parameters
+    ----------
+    area_2d : `~gammapy.irf.EffectiveAreaTable2D`
+        2D Effective Area Table
+    bcg_cube : `~gammapy.background.Cube`
+        Cube container class with scheme fixed to `bg_cube`
+    edisp_2d : `~gammapy.irf.EnergyDispersion2D`
+        Offset-dependent energy dispersion matrix
+    psf_mgauss : `~gammapy.irf.EnergyDispersion2D`
+        Position-dependent multi-Gauss PSF
+    """
+
+    def __init__(self, area_2d=None, bkg_cube=None, e_disp=None, psf_mgauss=None):
+        self.area_2d = area_2d
+        self.bkg_cube = bkg_cube
+        self.e_disp_2d = e_disp_2d
+        self.psf_mgauss = psf_mgauss
+
+    @classmethod
+    def from_fits(cls, irf_path):
+        """
+        Auxiliary constructor from a fits file
+
+        Parameters
+        ----------
+        path : `str`
+            Path of the file containing the IRF
+        """
+        # Dealing with effective area and fix headers
+        table_hdu_area = get_hdu(irf_path + '[EFFECTIVE AREA]')
+        table_area = Table(table_hdu_area.data)
+
+        table_area['ENERG_LO'].unit = u.TeV
+        table_area['ENERG_HI'].unit = u.TeV
+        table_area['THETA_LO'].unit = u.deg
+        table_area['THETA_HI'].unit = u.deg
+        table_area['EFFAREA'].unit = u.m * u.m
+        table_area['EFFAREA_RECO'].unit = u.m * u.m
+
+        area_2d = EffectiveAreaTable2D.from_table(table_area)
+
+        # Dealing with background and fix headers
+        table_hdu_bkg = get_hdu(irf_path + '[BACKGROUND]')
+
+        table_hdu_bkg.header['TTYPE7'] = 'Bgd'
+        table_hdu_bkg.header['TUNIT7'] = '1 / (MeV s sr)'
+
+        bkg_cube = Cube.from_fits_table(table_hdu_bkg, scheme='bg_cube')
+
+        # Dealing energy dispersion matrix
+        table_hdu_disp = get_hdu(irf_path + '[ENERGY DISPERSION]')
+        e_disp_2d = EnergyDispersion2D.from_fits(table_hdu_disp)
+
+        # Dealing with psf and fix values to get a single gaussian component
+        # by setting the amplitudes and the sigmas of the last two components
+        # to 0 and 1, respectively
+        table_hdu_psf = get_hdu(irf_path + '[POINT SPREAD FUNCTION]')
+
+        zero_array = np.zeros(shape=(2, 21), dtype='float32')
```

Comment:
Hard-coding a shape here is brittle, this will break with other IRFs.

Can't you just assign a scalar and let numpy broadcasting take care of it?
```
table_hdu_psf.data[0]['AMPL_2'] = 0
```
Alternatively, you could get the shape from an existing array like `table_hdu_psf.data[0]['AMPL_2'].shape`, no?